File: mdworkers/src/types/assetclass.rs
use core::fmt;

#[derive(Clone, Copy)]
pub enum AssetClass {
    Crypto,
    Forex,
    Equity,
}

impl fmt::Display for AssetClass {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            AssetClass::Crypto => write!(f, "crypto"),
            AssetClass::Forex => write!(f, "fx"),
            AssetClass::Equity => write!(f, "iex"),
        }
    }
}
impl fmt::Debug for AssetClass {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self)
    }
}

File: ws-server/internal/models/assetclass.go
package models

type AssetClass int

const (
	Forex AssetClass = iota
	Equity
	Crypto
)

var assetName = map[AssetClass]string{
	Forex:  "fx",
	Equity: "iex",
	Crypto: "crypto",
}

func (as AssetClass) String() string {
	return assetName[as]
}
